packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -539,6 +539,25 @@ class AbciAppDB:
 
     The parameters cleanup_history_depth and cleanup_history_depth_current can also be configured in skill.yaml so they are used automatically
     when the cleanup method is called from AbciApp.cleanup().
+
+    # Memory warning
+    -----------------------------------
+    The database is implemented in such a way to avoid indirect modification of its contents.
+    It copies all the mutable data structures\\*, which means that it consumes more memory than expected.
+    This is necessary because otherwise it would risk chance of modification from the behaviour side,
+    which is a safety concern.
+
+    The effect of this on the memory usage should not be a big concern, because:
+
+        1. The synchronized data of the agents are not intended to store large amount of data.
+         IPFS should be used in such cases, and only the hash should be synchronized in the db.
+        2. The data are automatically wiped after a predefined `cleanup_history` depth as described above.
+        3. The retrieved data are only meant to be used for a short amount of time,
+         e.g., to perform a decision on a behaviour, which means that the gc will collect them before they are noticed.
+
+    \\* the in-built `copy` module is used, which automatically detects if an item is immutable and skips copying it.
+    For more information take a look at the `_deepcopy_atomic` method and its usage:
+    https://github.com/python/cpython/blob/3.10/Lib/copy.py#L182-L183
```

Comment:
I did not add the warning log after all, because almost all of the methods of the DB's implementation are now using copies and the warnings would be logged too often. 

Instead, I appended the documentation of the class here, in order to inform the user how it works.